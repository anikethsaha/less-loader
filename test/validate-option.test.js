import fs from 'memfs';

import loader from '../src';

// eslint-disable-next-line consistent-return
const validate = (options) => {
  try {
    loader.call(
      Object.assign(
        {},
        {
          query: options,
          loaders: [],
          resolve: () => {},
          loadModule: () => {},
          fs,
          getResolve: () => () => {},
          async: () => (error) => {
            if (error) {
              throw error;
            }
          },
        }
      ),
      `
      @width: 10px;
      #header {
        width: @width;
      }
`
    );
    return 'Success';
  } catch (err) {
    return err.toString();
  }
};

const options = {
  failed: [{ wrong: 'options' }, { lint: 'shouldBeBool' }],
  success: [
    { color: true },
    { globalVars: '__Global_Primary' },
    { globalVars: ['__Global_Primary'] },
  ],
};

options.failed.forEach((option, index) => {
  it(`validate failed options #${index}`, () => {
    expect(validate(option)).toMatchSnapshot();
  });
});

options.success.forEach((option, index) => {
  it(`validate success options #${index}`, () => {
    expect(validate(option)).toBe('Success');
  });
});
