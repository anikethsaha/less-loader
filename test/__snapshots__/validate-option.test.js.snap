// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate failed options #0 1`] = `
"ValidationError: Invalid options object. LESS Loader has been initialised using an options object that does not match the API schema.
 - options has an unknown property 'wrong'. These properties are valid:
   object { lessOptions?, sourceMap? }"
`;

exports[`validate failed options #1 1`] = `
"ValidationError: Invalid options object. LESS Loader has been initialised using an options object that does not match the API schema.
 - options has an unknown property 'lint'. These properties are valid:
   object { lessOptions?, sourceMap? }"
`;

exports[`validate failed options #2 1`] = `"TypeError: Cannot create property 'plugins' on string 'string'"`;
